File: mobile/app/src/main/java/ru/morozovit/recipeviewer/mobile/Step.java
package ru.morozovit.recipeviewer.mobile;

import android.annotation.SuppressLint;
import android.widget.LinearLayout;
import android.widget.TextView;

import androidx.annotation.NonNull;

import com.google.android.material.textfield.TextInputLayout;

/** @noinspection DataFlowIssue*/
public class Step {
    public final LinearLayout root;

    public final TextView number;
    public final TextInputLayout description;

    @SuppressLint("SetTextI18n")
    public Step(@NonNull LinearLayout root, int number, String description) {
        this.root = root;
        this.number = (TextView) root.getChildAt(0);
        setNumber(number);
        this.description = (TextInputLayout) root.getChildAt(1);
        setDescription(description);
    }

    public Step(LinearLayout root) {
        this(root, 0, "");
    }

    public Step(LinearLayout root, int number) {
        this(root, number, "");
    }

    @SuppressLint("SetTextI18n")
    public void setNumber(int number) {
        this.number.setText("#" + number);
    }

    public int getNumber() {
        return Integer.parseInt(number.getText().toString().substring(1));
    }

    public String getDescription() {
        return description.getEditText().getText().toString();
    }

    public void setDescription(String description) {
        this.description.getEditText().setText(description);
    }
}

File: mobile/app/src/main/java/ru/morozovit/recipeviewer/mobile/Step.kt
package ru.morozovit.recipeviewer.mobile

import android.annotation.SuppressLint
import android.widget.LinearLayout
import android.widget.TextView
import com.google.android.material.textfield.TextInputLayout

@Suppress("UsePropertyAccessSyntax", "unused", "MemberVisibilityCanBePrivate")
@SuppressLint("SetTextI18n")
class Step(
    val root: LinearLayout,
    number: Int,
    description: String
) {
    private val number = root.getChildAt(0) as TextView
    private val description = root.getChildAt(1) as TextInputLayout

    constructor(root: LinearLayout): this(root, 0, "")
    constructor(root: LinearLayout, number: Int): this(root, number, "")

    init {
        setNumber(number)
        setDescription(description)
    }

    fun getNumber() = number.text!!
    fun getDescription() = description.editText!!.text!!

    fun setNumber(value: Int) = number.setText("#$value")
    fun setDescription(value: String) = description.editText!!.setText(value)
}